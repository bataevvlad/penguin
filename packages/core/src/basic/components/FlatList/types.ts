import {
  FlatListProps as DefaultFlatListProps,
  FlatList as DefaultFlatList,
} from 'react-native';

export type FlatListProps<T> = DefaultFlatListProps<T>;
export type IFlatList = DefaultFlatList;
