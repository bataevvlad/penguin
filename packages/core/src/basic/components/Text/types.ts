import {TextProps as DefaultTextProps, Text as DefaultText} from 'react-native';

export type ITextProps = DefaultTextProps;
export type IText = DefaultText;
