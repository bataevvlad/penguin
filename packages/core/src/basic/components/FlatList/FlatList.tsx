import {forwardRef} from 'react';
import {FlatList as DefaultFlatList} from 'react-native';

import {FlatListProps, IFlatList} from './types';

// eslint-disable-next-line react/display-name
const FlatList = forwardRef<IFlatList, FlatListProps<never>>((props, ref) => {
  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
  // @ts-ignore
  return <DefaultFlatList ref={ref} {...props} />;
});
export default FlatList;
