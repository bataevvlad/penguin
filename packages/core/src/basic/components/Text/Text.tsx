import {forwardRef} from 'react';
import {Text as DefaultText} from 'react-native';

import {getConfig} from '../../config';

import {IText, ITextProps} from './types';

// eslint-disable-next-line react/display-name
const Text = forwardRef<IText, ITextProps>((props, ref) => {
  return (
    <DefaultText
      ref={ref}
      allowFontScaling={getConfig().allowFontScaling}
      {...props}
    />
  );
});
export default Text;
