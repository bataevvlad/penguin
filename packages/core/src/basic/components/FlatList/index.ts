import FlatList from './FlatList';

export * from './types';

export {FlatList};
