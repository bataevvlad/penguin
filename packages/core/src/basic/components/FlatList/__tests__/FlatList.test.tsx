import {render} from '@testing-library/react-native';

import FlatList from '../FlatList';
import Text from '../../Text/Text';

describe('@penguin/core/FlatList', () => {
  it('renders correctly', () => {
    const renderItem = ({item}: {item: string}) => {
      return <Text>{item}</Text>;
    };
    const {toJSON} = render(
      <FlatList
        data={['1', '2', '3']}
        renderItem={renderItem}
        keyExtractor={(item: string, index: number) =>
          item.toString() + index.toString()
        }
      />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
