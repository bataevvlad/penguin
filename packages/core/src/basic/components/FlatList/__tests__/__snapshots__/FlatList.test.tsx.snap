// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@penguin/core/FlatList renders correctly 1`] = `
<RCTScrollView
  data={
    [
      "1",
      "2",
      "3",
    ]
  }
  getItem={[Function]}
  getItemCount={[Function]}
  keyExtractor={[Function]}
  onContentSizeChange={[Function]}
  onLayout={[Function]}
  onMomentumScrollBegin={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  removeClippedSubviews={false}
  renderItem={[Function]}
  scrollEventThrottle={0.0001}
  stickyHeaderIndices={[]}
  viewabilityConfigCallbackPairs={[]}
>
  <View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        allowFontScaling={true}
      >
        1
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        allowFontScaling={true}
      >
        2
      </Text>
    </View>
    <View
      onFocusCapture={[Function]}
      onLayout={[Function]}
      style={null}
    >
      <Text
        allowFontScaling={true}
      >
        3
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
